src/group_theory/perm/cycle/basic.lean

```diff
@@ -1609,3 +1611,85 @@ end
 end fixed_points
 
 end equiv.perm
+
+open equiv
+
+namespace set
+variables {f : perm α} {s : set α}
+
+lemma prod_self_eq_Union_perm (hf : f.is_cycle_on s) :
```

Comment:
The `n : ℤ` is a bit surprising to me cardinality-wise. Just to be sure, you have examples of finite and uncountably infinite sets with `is_cycle_on`, right?

Reply:
`s.is_cycle_on f` implies `countable f` because of the `ℤ` in the definition of `is_cycle_on`. Finite sets are taken care of by `finset.product_self_eq_disj_Union_perm`

Reply:
I see, I was thinking of `succ` being a cycle on uncountable ordinals, but of course that doesn't work for limits. So LGTM.